OpenEdXMobile/src/main/java/org/edx/mobile/base/MainApplication.java

```diff
@@ -140,13 +140,14 @@ private void init() {
         }
 
         // Braze SDK Initialization
-        if (config.getBrazeConfig().isEnabled()) {
+        if (config.getBrazeConfig().isEnabled() && config.getFirebaseConfig().isEnabled()) {
             AppboyConfig appboyConfig = new AppboyConfig.Builder()
-                    .setApiKey(config.getBrazeConfig().getApiKey())
-                    .setCustomEndpoint(config.getBrazeConfig().getEndPointKey())
+                    .setIsFirebaseCloudMessagingRegistrationEnabled(config.getBrazeConfig().isPushNotificationsEnabled())
```

Comment:
IMO following is better cuz if the firebase notifications are not enabled then need to check the braze notification parameter.
```
.setIsFirebaseCloudMessagingRegistrationEnabled(config.areFirebasePushNotificationsEnabled() && config.getBrazeConfig().isPushNotificationsEnabled())
```